src/jaxsim/parsers/descriptions/joint.py

```diff
@@ -14,39 +13,35 @@
 from .link import LinkDescription
 
 
-@enum.unique
-class JointType(enum.IntEnum):
-    """
-    Type of supported joints.
-    """
-
-    @staticmethod
-    def _generate_next_value_(name, start, count, last_values):
-        # Start auto Enum value from 0 instead of 1
-        return count
-
-    #: Fixed joint.
-    F = enum.auto()
+class _JointTypeMeta(type):
+    def __new__(cls, name, bases, dct):
+        cls_instance = super().__new__(cls, name, bases, dct)
 
-    #: Revolute joint (1 DoF around axis).
-    R = enum.auto()
+        # Assign integer values to the descriptors
+        cls_instance.F = 0
+        cls_instance.R = 1
+        cls_instance.P = 2
 
-    #: Prismatic joint (1 DoF along axis).
-    P = enum.auto()
+        return cls_instance
 
 
-@jax_dataclasses.pytree_dataclass
-class JointDescriptor:
+class JointType(metaclass=_JointTypeMeta):
```

Comment:
What is the advantage of using such pattern based on metaclasses (that often confuse even experience Python developers) over a much more simple approach as the following?

```python
from typing import ClassVar

@dataclasses.dataclass(frozen=True)
class JointType:
    F: ClassVar[int] = 0
    R: ClassVar[int] = 1
    P: ClassVar[int] = 2
```

Reply:
Since the same logic can be applied to substitute the `enum` of `VelRepr` I thought this could be easily reusable in the future. Nonetheless, I agree with you that the dataclasses implementation is more straightforward and should have similar behavior

Reply:
Correct! The usage of metaclasses can be often beneficial when they can be reused. In this case, the metaclass is regardless something specific to the joint type, and cannot be reused elsewhere. Probably using the combination of dataclasses and ClassVar also in this case for joint types would result to a much more readable code (I tested the approach myself and tests pass).

Reply:
Perfect, thanks for the suggestion. Addressed in https://github.com/ami-iit/jaxsim/pull/153/commits/f774b3d0030573912e3f2f969539af91448a7762